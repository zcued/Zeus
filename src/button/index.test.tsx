import * as React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import Button from '.'

test('it works', () => {
  const wrapper = shallow(<Button />)
  expect(wrapper).toMatchSnapshot()
})

describe('button works', () => {
  it('should render an `.icon`', () => {
    const wrapper = shallow(<Button icon="close">button</Button>)
    expect(wrapper.find('Icon')).toHaveLength(1)
  })

  it('should render a `spin`', () => {
    const wrapper = shallow(<Button icon="close" loading={true} />)
    expect(wrapper.prop('disabled')).toEqual(true)
  })
  
  it('should render a child', () => {
    const children = <i className="test" />
    const wrapper = shallow(<Button icon="close" children={children} />)
    expect(wrapper.find('.test').length).toEqual(1)
  })
})
