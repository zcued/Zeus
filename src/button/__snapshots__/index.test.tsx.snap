// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        null,
        null,
      ],
      "disabled": undefined,
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          null,
          null,
        ],
        "disabled": undefined,
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
